dpctl-capi/include/dpctl_sycl_context_interface.h

```diff
@@ -99,6 +99,16 @@ DPCTL_API
 __dpctl_give DPCTLSyclContextRef
 DPCTLContext_Copy(__dpctl_keep const DPCTLSyclContextRef CRef);
 
+/*!
+ * @brief Returns the number of devices associated with sycl::context referenced
+ * by DPCTLSyclContextRef object.
+ *
+ * @param    CRef           DPCTLSyclContexRef object to query.
+ * @return   A positive count on success or zero on error.
+ */
+DPCTL_API
+size_t DPCTLContext_DeviceCount(__dpctl_keep const DPCTLSyclContextRef CRef);
```

Comment:
Why not expose the `context->get_devices()` directly and return a `DPCLDeviceVectorRef`?

Reply:
`DPCTLContext_GetDevices` does that.